### PR TITLE
[CLOUD-390] Docker upgrade from v19.03 to v23.0.1-1

### PR DESCRIPTION
**Pull request by gibchikafa**

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -493,18 +493,15 @@
...
+default['hops']['docker_version']['centos']           = "20.10.9-3"
+default['hops']['docker_version']['ubuntu']           = "20.10.7"
 default['hops']['docker']['group_id']                 = '1513'
 default['hops']['docker']['storage_driver']           = "overlay2"
 
 
-default['hops']['docker']['pkg']['download_url']['centos'] ="#{node['download_url']}/docker/#{node['hops']['docker_version']['centos']}/rhel"
-default['hops']['docker']['pkg']['download_url']['ubuntu'] ="#{node['download_url']}/docker/ubuntu/#{node['hops']['docker_version']['ubuntu'].split('-')[0]}/"
+
+default['hops']['docker']['download_url']['centos'] = "https://repo.hops.works/dev/gibson/docker/20.10.9-3.tgz"
```
Comment (SirOibaf) [inline attributes/default.rb:503]: Why did you change this? - it's available here and it works: https://repo.hops.works/master/docker/rhel/

File: metadata.rb [hunk 2]
```diff
@@ -882,14 +882,6 @@
...
 
-attribute "hops/selinux_version/centos",
-          :description =>  "the version of selinux to use on centos installation",
-          :type => 'string'
-
-attribute "hops/containerd_version/centos",
-          :description =>  "the version of containerd to use on centos installation",
-          :type => 'string'
-
 attribute "hops/containerd_version/ubuntu",
```
Comment (SirOibaf) [inline metadata.rb:885]: Can you remove the other variables as well? Like this one, but there are probably more that we are not using anymore.

**Review by SirOibaf — CHANGES_REQUESTED** [review]

File: attributes/default.rb (hunk 1 above)
Comment (gibchikafa) [reply attributes/default.rb:503]: I updated containerd to the latest version 1.6.9. There was some issue with the older containerd you have, but I forgottent the error. I can try to reproduce it later.

File: recipes/docker.rb [hunk 3]
```diff
@@ -95,6 +88,19 @@
...
+  only_if { File.exist? '/etc/containerd/config.toml' }
+end
+
+bash "restart_containerd" do
+  user 'root'
+  group 'root'
+  code <<-EOH
+      systemctl restart containerd
+  EOH
+end
```
[the remark this answers is not in the document]
Comment (gibchikafa) [reply recipes/docker.rb:102]: The default config.toml has `CRI` plugin disabled then cant work with kubernetes 1.26.1.

[the remark this answers is not in the document]
Comment (SirOibaf) [reply recipes/docker.rb:102]: Ok but then can you moved this operation on the kube-hops-chef repo? I would like to keep a stock configuration if we don't have K8s installed on the machines.

[the remark this answers is not in the document]
Comment (gibchikafa) [reply recipes/docker.rb:102]: Alright

**Review by SirOibaf — APPROVED** [review]
